commons/src/main/java/org/infinispan/commons/api/AsyncLifecycle.java

```diff
@@ -0,0 +1,16 @@
+package org.infinispan.commons.api;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Async Lifecycle interface that defines the lifecycle of components in a non blocking way
+ *
+ * @author Katia Aresti (<a href="mailto:karesti@redhat.com">karesti@redhat.com</a>)
+ * @since 9.1
+ */
+public interface AsyncLifecycle {
```

Comment:
Why do you create the interface when the lifecycle is managed by the user anyway?